func/omp/simple_atomic.cpp

```diff
@@ -0,0 +1,32 @@
+#include <cstdio>
+#include <omp.h>
+
+#include <faasm/shared_mem.h>
+
+int main()
+{
+    int nThreads = 100;
+    int counter = 20;
+
+#pragma omp parallel num_threads(nThreads) default(none) shared(counter)
```

Comment:
I am not sure I understand what this test is doing. If we are testing the atomic pragma, I would expect to see the
```cpp
#pragma omp atomic
```
construct.

I assume this has to do with the `NOTE` beneath, but I still don't really get it.

Reply:
If we were compiling to native code we would have `#pragma omp atomic`, but because this doesn't actually result in anything we can hook into in Faasm, we have to use the custom Faasm macro.